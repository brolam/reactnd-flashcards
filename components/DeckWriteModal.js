import React from 'react'
import { StyleSheet } from 'react-native'
import {
  Modal,
  View,
  Text,
  TextInput,
  TouchableOpacity,
  KeyboardAvoidingView,
  Platform
} from 'react-native'
import CardStyle from '../styles/cardStyles'
import { white, green, red } from '../styles/colors'

export default function DeckWriteModal(props) {
  return (
    <Modal animationType='fade' transparent={true} >
      {Platform.OS === 'ios' ?
        <KeyboardAvoidingView behavior='padding' style={styles.container}>
          <DeckWriteBody {...props} />
        </KeyboardAvoidingView>
        :
        <View style={styles.container} >
          <DeckWriteBody {...props} />
        </View>
      }
    </Modal>
  )
}

function DeckWriteBody(props) {
  return (
    <View style={[CardStyle, styles.body]}>
      <Text style={styles.title}>{props.title}</Text>
      <TextInput
        placeholder="Enter with a Deck Title."
        style={styles.titleTextInput}
        onChange={() => { }} />
      <View style={styles.buttonsContainer}>
        <TouchableOpacity
          style={[buttonStyle, styles.buttonSave]}
          onPress={() => console.log('Save Deck')}>
          <Text style={styles.textButtons}>Save</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[buttonStyle, styles.buttonCancel]}
          onPress={() => console.log('Cancel Add Deck')}>
          <Text style={styles.textButtons}>Cancel</Text>
        </TouchableOpacity>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.5)',
  },
  body: {
    flex: 0,
    flexDirection: 'column',
    alignItems: 'center',
    width: (Platform.OS !== 'ios' ? '80%' : undefined)
  },
  title: {
    fontSize: 18,
    marginBottom: 10,
    fontWeight: '600',
  },
  titleTextInput: {
    fontSize: 24,
    marginBottom: 20,
    height: 50,
    width: '100%'
  },
  buttonsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    flexWrap: 'wrap',
    marginBottom: 5
  },
  buttonSave: {
    backgroundColor: green,
    margin: 5
  },
  buttonCancel: {
    backgroundColor: red,
    margin: 5
  },
  textButtons: {
    color: white,
    fontSize: 24
  },
});