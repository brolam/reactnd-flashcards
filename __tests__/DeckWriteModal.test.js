import React from 'react'
import DeckWriteModal from '../components/DeckWriteModal'

test('renders without crashing', () => {
  const wrapper = shallow(<DeckWriteModal />);
  expect(wrapper).toMatchSnapshot();
});

test('onSave event', () => {
  const onSave = jest.fn()
  const deckWriteModal = mount(<DeckWriteModal onSave={onSave} />);
  const textInputTitle = deckWriteModal.find('TextInput').at(0)
  const buttonSave = deckWriteModal.find('TouchableOpacity').at(0)
  textInputTitle.instance().value = 'One Deck'
  buttonSave.props().onPress()
  expect(onSave).toHaveBeenCalled();
});

describe('save new Deck', () => {
  let deckWriteModal
  let onSave
  beforeEach(() => {
    onSave = jest.fn()
    deckWriteModal = mount(<DeckWriteModal onSave={onSave} />);
  })

  test('title is required', () => {
    const buttonSave = deckWriteModal.find('TouchableOpacity').at(0)
    buttonSave.props().onPress()
    expect(onSave).not.toHaveBeenCalled();
  });

})