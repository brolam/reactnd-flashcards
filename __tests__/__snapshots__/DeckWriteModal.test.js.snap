// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing - android 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <DeckWriteBody />
  </View>
</Component>
`;

exports[`renders without crashing 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <KeyboardAvoidingView
    behavior="padding"
    keyboardVerticalOffset={0}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <DeckWriteBody />
  </KeyboardAvoidingView>
</Component>
`;
